File: tests/benchmarks/resolution/fixtures/fsharp/Service.fs
module Service

let validateUser name email age =
    Validators.validateName name
    && Validators.validateEmail email
    && Validators.validateAge age

let createUser uid name email age =
    if validateUser name email age then
        Repository.save uid { Repository.Name = name; Repository.Email = email; Repository.Age = age }
        Ok ()
    else
        Error "Validation failed"

let getUser uid =
    Repository.findById uid

let removeUser uid =
    Repository.delete uid

let summary () =
    Repository.count ()

File: tests/benchmarks/resolution/fixtures/ocaml/service.ml
let validate_user name email age =
  Validators.validate_name name
  && Validators.validate_email email
  && Validators.validate_age age

let create_user uid name email age =
  if validate_user name email age then begin
    Repository.save uid { Repository.name; email; age };
    Ok ()
  end else
    Error "Validation failed"

let get_user uid =
  Repository.find_by_id uid

let remove_user uid =
  Repository.delete uid

let summary () =
  Repository.count ()
